file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/business/impl/ItemsActionBusinessImpl.java

```diff
@@ -129,13 +129,48 @@ public FileItem saveFileItem(UploadFileAction uploadAction) {
 		fileItem.setPath(uploadAction.getPath());
 		fileItem.setUploadTime(Timestamp.from(Instant.now()));
 		fileItem.setUser(usersDao.find(uploadAction.getUserId()).orElse(null));
-		
-		fileItemDao.save(fileItem);
+		fileItem.setSize(uploadAction.getSize());
+
+        if(fileItemDao.exists(fileItem)) {
+            List<String> fileItems = fileItemDao.getSimilarFileItemsFromSameDirectory(fileItem);
+            fileItem.setName(getDuplicateNameWithSuffix(fileItem.getName(), getNextAvailableSuffix(fileItems)));
+        }
+
+        fileItemDao.save(fileItem);
 		fileItemDao.flush();
+
 		return fileItem;
 	}
 
-	// -----------------------------------------------------------------
+	String getDuplicateNameWithSuffix(String fileName, String suffix) {
```

Comment:
We can move all the logic of duplicate naming into a separate component. In this way, we keep the single responsibility principle and can test the logic in isolation.